azurelinuxagent/common/protocol/restapi.py

```diff
@@ -180,9 +183,11 @@ def __init__(self, name=None):
         self.versionUris = DataContractList(ExtHandlerVersionUri)
 
     def sort_key(self):
-        level = self.properties.dependencyLevel
-        if level is None:
+        levels = [e.dependencyLevel for e in self.properties.extensions]
+        if len(levels) == 0:
             level = 0
+        else:
+            level = min(levels)
```

Comment:
Min or max?
>  An extension's distance from the roots of the dependency graph is based on the maximum number of edges between that extension and any of the roots of the acyclic dependency graph.

Reply:
This is for the handler which needs to be installed before any of its extensions may be enabled. Right now the agent only supports one extension per handler, but I think `min` makes sense for a handler with more extensions. Additional changes would be needed to support more than one extension per handler but that is out of scope for this PR.

Reply:
That's not what your spec says, which I quoted. Assuming I understand it right.

Reply:
OK,  your in-office conversation with me convinced me. This is talking about the level at which the extension handler gets "handled", and it has to be done before the shallowest dependent extension gets "handled", and that means "min". Got it.